conda/notices/core.py

```diff
@@ -124,10 +124,19 @@ def wrapper(*args, **kwargs):
                 logger.error(f"Unable to open cache file: {str(exc)}")
 
             if channel_notice_set is not None:
-                return_value = func(*args, **kwargs)
-                display_notices(channel_notice_set)
-
-                return return_value
+                try:
+                    return_value = func(*args, **kwargs)
+                    display_notices(channel_notice_set)
+
+                    return return_value
+
+                except Exception:
+                    try:
+                        # Remove the notices cache file if we encounter an exception
+                        cache.clear_cache()
+                    except OSError:
```

Comment:
Hm maybe `clear_cache` should handle `OSError` for us?

Reply:
I would say no because it should be up to the calling code to decide whether the exception can be ignored or not.